source/adaptive_load/input_variable_setter_impl.cc

```diff
@@ -37,6 +37,11 @@ InputVariableSetterPtr RequestsPerSecondInputVariableSetterConfigFactory::create
   return std::make_unique<RequestsPerSecondInputVariableSetter>(config);
 }
 
+absl::Status RequestsPerSecondInputVariableSetterConfigFactory::ValidateConfig(
+    const Envoy::Protobuf::Message&) const {
+  return absl::OkStatus();
```

Comment:
See new comment in config_validator.h that explains the current contract here.  The full story  is a bit complicated and not ideal because of the mixture of these weak types, exceptions, and Status.

The short answer is that I intentionally don't want the ValidateConfig author to be responsible for checking the type because it's already checked elsewhere.

Also adding a comment to issue #468 about potential template gymnastics to make factories more explicit about types.

What do you think?